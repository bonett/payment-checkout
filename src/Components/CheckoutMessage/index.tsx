/* eslint-disable */
import React from "react";
import { moneyFormatter } from "Utils";

function CheckoutMessage({
  message,
  payment,
}: {
  message: string;
  payment?: any;
}) {
  return (
    <div className="message-content">
      <h2>{message}</h2>
      <>
        {payment && (
          <div className="message-content_wrapper">
            <p>
              You have bought 1 {payment?.product} <br /> for{" "}
              {moneyFormatter(payment?.price, "COP")} <br /> <br /> 
              <strong>Thank you, Enjoy it !</strong>
            </p>
          </div>
        )}
      </>
    </div>
  );
}

export default CheckoutMessage;
