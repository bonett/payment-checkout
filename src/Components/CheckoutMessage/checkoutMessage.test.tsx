import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import CheckoutMessage from "./";
import store from "Store";

describe("CheckoutMessage component", () => {
  test("Should renders CheckoutMessage component", () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CheckoutMessage
            message={"Payment successfully"}
            payment={{ product: "Iphone 12", price: 12321 }}
          />
        </BrowserRouter>
      </Provider>
    );

    const titleSucess = screen.getByText(/Payment successfully/i);
    const productName = screen.getByText(/Iphone 12/i);
    expect(titleSucess).toBeInTheDocument();
    expect(productName).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("Should renders error in component", () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CheckoutMessage message={"Error, something was wrong"} />
        </BrowserRouter>
      </Provider>
    );

    const titleSucess = screen.getByText(/Error, something was wrong/i);
    expect(titleSucess).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
