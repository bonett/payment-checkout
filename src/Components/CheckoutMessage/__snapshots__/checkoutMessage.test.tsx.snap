// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutMessage component Should renders CheckoutMessage component 1`] = `
<div>
  <div
    class="message-content"
  >
    <h2>
      Payment successfully
    </h2>
    <div
      class="message-content_wrapper"
    >
      <p>
        You have bought 1 
        Iphone 12
         
        <br />
         for
         
        $Â 12.321
         
        <br />
         
        <br />
        <strong>
          Thank you, Enjoy it !
        </strong>
      </p>
    </div>
  </div>
</div>
`;

exports[`CheckoutMessage component Should renders error in component 1`] = `
<div>
  <div
    class="message-content"
  >
    <h2>
      Error, something was wrong
    </h2>
  </div>
</div>
`;
